src/hooks/useAppState/reducer.js

```diff
@@ -1,23 +1,30 @@
 export const types = {
-  SET_LIST_ID: 'SET_LIST_ID',
+  NEW_LIST: 'NEW_LIST',
   SET_STATE: 'SET_STATE',
   NEW_ITEM: 'NEW_ITEM',
   DELETE_ITEM_BY_KEY: 'DELETE_ITEM_BY_KEY',
   UPDATE_ITEM_BY_KEY: 'UPDATE_ITEM_BY_KEY',
   TOGGLE_COMPLETION_BY_KEY: 'TOGGLE_COMPLETION_BY_KEY',
   STORE_ID: 'STORE_ID',
+  STORE_LIST_ID: 'STORE_LIST_ID',
   MERGE_ITEMS: 'MERGE_ITEMS',
 };
 
-export const initialState = { items: [], listId: null };
+export const initialState = { items: [], lists: [], index: 0 };
```

Comment:
The new shape of the state is ([flow](https://flow.org/en/docs/types/objects/) syntax):
```js
type state = {
  items: Array<Array<{ // this is a two dimensional array, the first dimension is which list we have, and the second is each item in that list.
    id: string, // each item has these four properties
    text: string,
    key: string,
    isCompleted: boolean,
  }>>,
  lists: Array<{
    id: string,  // each list has these three properties
    key: string,
    name: string,
  }>,
  index: number, // this tells us which list is selected.
}